broadinstitute/ddp-study-server
# [DSM] DDP-8678: exact match problem

## Context

Ticktet: https://broadinstitute.atlassian.net/browse/DDP-8678

ElasticSearch matching query has `operator` property which is either `AND` or `OR`, if we would like to have exact match we should set `AND` operator to matching query, otherwise `OR`, since we did not consider this detail while ago, we had problem with exact search.

**Things have done:**

- Add method to `MatchingQueryStrategy` to set `operator` either `AND` or `OR`.

**How to test:**

- Go to participant list page, choose any field to filter and check `Exact Match`, returned result should return only records which matches exactly to the chosen field's value.

## Checklist

- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :sweat_smile: There might be some issues here

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [ ] I have written zero automated tests but have poked around locally to verify proper functionality

## Release
